src/ed25519.rs

```diff
@@ -129,7 +129,7 @@ impl SecretKey {
     /// A `SecretKey`.
     #[inline]
     #[allow(dead_code)]
-    fn from_bytes(bytes: &[u8]) -> SecretKey {
+    pub fn from_bytes(bytes: &[u8]) -> SecretKey {
```

Comment:
I imagine you'll have to do something like this eventually, otherwise there's presently no way to load an existing secret key, only generate a random one.

I believe in @briansmith's *ring* library he recomputes the Ed25519 public key from the private key to ensure it matches (and therefore the keypair is valid)